### PR TITLE
[quick video overview]  Add epoch_key to CometLogger to make epoch extraction configurable

### PR DESCRIPTION
**Pull request by pupeno**
## What does this PR do?

Contributes to #21782

I made a quick video overview that might save you some time during review:: https://youtu.be/xly3bs5VOzs

Adds `epoch_key` to `CometLogger` to let users control which incoming metric key is extracted
and forwarded to Comet's dedicated `epoch=` argument.

Previously this key was hardcoded to `"epoch"`. With this change it is configurable, and can be
set to `None` to disable epoch extraction entirely.

Default is `"epoch"`, which preserves the previous behaviour — fully backward-compatible.

```python
# before: always extracted "epoch" from metrics
logger = CometLogger(...)

# after: same default, no change
logger = CometLogger(...)

# extract a prefixed epoch key (e.g. when used with Trainer(log_key_prefix="trainer/"))
logger = CometLogger(epoch_key="trainer/epoch")

# disable epoch extraction entirely
logger = CometLogger(epoch_key=None)
```

Sample code can be found on https://github.com/pupeno/pytorch-lightning-workspace/tree/main/21786_comet_epoch_key

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

**Comment by pupeno** [comment]
I updated this to master and there's sample code in https://github.com/pupeno/pytorch-lightning-workspace/blob/main/scratch/21786/experiment.py

Anything else I can do here?
